File: codes/aoj/itp1/itp1_7_b.cpp
#include <bits/stdc++.h>
using namespace std;

int solve(int n, int x) {
    int res = 0;
    for (int i = 1; i <= n; i++) {
        for (int j = i + 1; j <= n; j++) {
            for (int k = j + 1; k <= n; k++) {
                if (i + j + k == x) {
                    res++;
                }
            }
        }
    }

    return res;
}

int main() {
    int n, x;
    while (true) {
        cin >> n >> x;
        if (n == 0 && x == 0) {
            break;
        }
        cout << solve(n, x) << "\n";
    }
}

File: aoj/itp1_7_b.py
def solve(n: int, x: int):
    res = 0
    for i in range(1, n + 1):
        for j in range(i + 1, n + 1):
            k = x - i - j
            if k > j and k <= n:
                res += 1
    return res


def main():
    while 1:
        n, x = map(int, input().split())
        if n == 0 and x == 0:
            break
        result = solve(n, x)
        print(result)


if __name__ == "__main__":
    main()
